src/preferences/common/preference-protocol.ts

```diff
@@ -12,9 +12,8 @@ export const preferencesPath = '/services/preferences';
 export const PreferenceServer = Symbol("PreferenceServer")
 
 export interface PreferenceServer extends Disposable {
-    has(preferenceName: string): Promise<boolean>;
-    get<T>(preferenceName: string): Promise<T | undefined>;
     setClient(client: PreferenceClient | undefined): void;
+    onReady(): Promise<void>;
```

Comment:
I was thinking to do it is not through a request, but via the existing notification:
```ts
export interface PreferenceClient {
     onDidChangePreference(event: PreferenceChangedEvent): void
 }

export interface PreferenceChangedEvent {
     changes: PreferenceChange[]
}
export interface PreferenceChange {
     readonly preferenceName: string;
     readonly newValue?: any;
     readonly oldValue?: any;
}
```
then `PreferenceService` is ready only when it is received at least one notificaiton from the server.